cmd/packer-sdc/internal/fix/gocty.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package fix
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"golang.org/x/mod/modfile"
+)
+
+const (
+	oldPath    string = "github.com/zclconf/go-cty"
+	newPath    string = "github.com/nywilken/go-cty"
+	newVersion string = "1.12.1"
+)
+
+type GoCtyFix struct {
+	OldPath, NewPath, NewVersion string
+	data                         []byte
+}
+
+// NewGoCtyFixer is an entry to the go-cty fix command
+func NewGoCtyFixer() GoCtyFix {
+	return GoCtyFix{
+		OldPath:    oldPath,
+		NewPath:    newPath,
+		NewVersion: newVersion,
+	}
+}
+
+func (f GoCtyFix) modFileFormmatedVersion() string {
+	return fmt.Sprintf("v%s", f.NewVersion)
+}
+
+func (f GoCtyFix) isUnfixed(dir string) (bool, error) {
+	modFilePath := filepath.Join(dir, "go.mod")
+	if _, err := os.Stat(modFilePath); err != nil {
+		return false, errors.Wrap(err, "failed to find go.mod file")
+	}
+	data, err := os.ReadFile(modFilePath)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to read go.mod file")
+	}
+	mf, err := modfile.Parse(modFilePath, data, nil)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse go.mod file")
+	}
+
+	// Basic go.mod with no module dependencies
+	if len(mf.Require) == 0 {
+		return false, nil
+	}
+
+	// Brute force will be better to Sort then search
+	var found bool
+	for _, req := range mf.Require {
+		if req.Mod.Path == f.OldPath {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return false, nil
+	}
+
+	if len(mf.Replace) == 0 {
+		return true, nil
+	}
+	// what happens with multiple replace
+	for _, r := range mf.Replace {
+		if r.Old.Path != f.OldPath {
+			continue
+		}
+		if r.New.Path != f.NewPath {
+			return false, errors.New("found unexpected replace for " + r.Old.Path)
+		}
+		return r.New.Version != f.modFileFormmatedVersion(), nil
+	}
+	return true, nil
+}
+
+func (f GoCtyFix) Fix(dir string, check bool) error {
+	ok, err := f.isUnfixed(dir)
```

Comment:
While I understand the intent, I'm not sure `ok` is the best name for this, maybe consider naming it `unfixed` to make it clear what it is we're testing on?

Reply:
Good call out. I made the suggested change.